### PR TITLE
feat(query-rules): add query-rules support

### PR DESCRIPTION
**Pull request by rayrutjes**
I expect the tests to fail on travis given the feature is not deployed on our testing APP ID.

### CODE REVIEW

**Comment by clement-leprovost** [comment]
@rayrutjes The API is deployed everywhere: we can merge it.

**Comment by rayrutjes** [comment]
Waiting for the tests to pass and for an additional review.

**Comment by rayrutjes** [comment]
@julienbourdeau @maxiloc @JanPetr 
Can any of you give this a review?

File: src/AlgoliaSearch/Index.php [hunk 1]
```diff
@@ -1506,6 +1506,142 @@ public function searchFacet($facetName, $facetQuery, $query = array())
...
+    public function saveRule($objectID, $content, $forwardToReplicas = false)
+    {
+        if (!isset($content['objectID'])) {
+            $content['objectID'] = $objectID;
+        }
+
+        return $this->client->request(
+            $this->context,
+            'PUT',
+            '/1/indexes/'.$this->urlIndexName.'/rules/'.urlencode($objectID).'?forwardToReplicas='.($forwardToReplicas ? 'true' : 'false'),
```
Comment (julienbourdeau) [inline src/AlgoliaSearch/Index.php]: I feel like if `$content['objectID']` is different from `$objectID` we might be updating the wrong rule. I'm not sure I understand why we need to pass an $objectID.

Comment (clement-leprovost) [inline src/AlgoliaSearch/Index.php]: The REST API requires the `objectID` to be part of the URL *and* also part of the payload. I know it's not very RESTful, but it's the way it works at the moment. Don't worry, the API will reject the request if the two object IDs mismatch.

**Comment by clement-leprovost** [comment]
@julienbourdeau `saveRule` is not a partial update, it's a full update. It's called "save" because it can be either a create or an update (PUT method).

Note: The API for rules is essentially the same as that for [synonyms](https://www.algolia.com/doc/api-client/php/synonyms/).

**Pull request merged**